Spellcheck spans

sbaughman opened the pull request.

sbaughman committed:
Remove unused variable

sbaughman committed:
Remove markTypos and markAllTypos functions

Not used anymore now that all spellchecks are done at block level

sbaughman committed:
Remove more unused variables

sbaughman committed:
Make word walker ignore spellcheck spans

sbaughman committed:
Stop removing all spellcheck spans on every render

caffodian commented on plugins/nanospell/plugin.js:
```diff
@@ -887,12 +885,6 @@
-			range.selectNodeContents(node);
-
-			this.markTyposInRange(editor, range);
-		},
```
I'm surprised that this could go away.  Then I noticed that in scanWords, we duplicate this functionality.

caffodian replied:
I think I'd rather keep this method and then have scanWords call it instead.  Mostly for clarity reasons, and also, I suspect that in the future, we may have some cases where we'll want to be smarter about doing element checking vs. actual range checking.  (For example, copy/paste or drag and drop, which don't happen as often, but we haven't really thought about at all)

caffodian commented on plugins/nanospell/plugin.js:
```diff
@@ -919,17 +911,6 @@
-			while (( block = iterator.getNextParagraph() )) {
-				this.markTypos(editor, block);
-			}
-		},
```
Ah yes.  I noticed this function was dead code as well, after the previous thing.

caffodian reviewed (commented):
looks good to me, pending test revisions

caffodian committed:
Merge pull request #15 from PolicyStat/word_walker

Prevent words from getting mashed together when walking nested lists

sbaughman committed:
Merge branch 'pstat_rewrite' of github.com:PolicyStat/ckeditor-spell-check-plugin-js-dev-edge into spellcheck_spans

sbaughman committed:
Add markTypos back in

not currently used, but will likely be needed in the future.  Still
leaving markAllTypos out, as it is not needed now that everything is
done by block-level

sbaughman committed:
Refactor render function to use markTypos

sbaughman committed:
Move event handler setup so they are all together

sbaughman committed:
Remove outdated comment

sbaughman committed:
refactor spellCheckInProgress

sbaughman committed:
Add checks to wordwalker

before returning an object when ‘WordWalker.getNextWord’ is called,
check to make sure that the word is not an empty string

sbaughman committed:
Add test to verify spellcheck spans are added when a spellcheck occurs

sbaughman committed:
Add test to verify the word walker skips spellcheck spans

sbaughman committed:
Ignoring obsolete tests

‘markAllTypos’ function was removed after rewriting spellchecker to
check by block.  Will find a way to test these cases elsewhere

caffodian commented on plugins/nanospell/plugin.js:
```diff
@@ -267,6 +254,9 @@
 			});
+			editor.on(EVENT_NAMES.START_RENDER, render, self);
+			editor.on(EVENT_NAMES.START_SCAN_WORDS, scanWords, self);
+			editor.on(EVENT_NAMES.START_CHECK_WORDS, checkWords, self);
```
curious why you moved these.  Not opposed.  I was floating between these two options for a while.

sbaughman replied:
Just seemed to make sense to have all the event listener setup together.  I also like the pattern of making a `setupEventListeners` function to group them together and calling that from `init`

sbaughman committed:
Add conditions to handle line breaks and other elements that should be word boundaries.

caffodian commented on plugins/nanospell/plugin.js:
```diff
@@ -96,12 +96,16 @@
 				 	((blockLimit && !blockLimitIsStartNode) ||
-					(block && !blockIsStartNode))) {
+					(block && !blockIsStartNode) ||
+					node.is('br') || node.is('sup') || node.is('sub'))) {
```
we can actually do

`node.is('br', 'sup', 'sub')`

at least this is what http://docs.ckeditor.com/#!/api/CKEDITOR.dom.element-method-is would indicate.

caffodian reviewed (changes requested):
Could we also add a simple walking test, maybe something like

`<p>sometext<br/>something<sup>superscript</sup> anotherthing<sub>subscript</sub>` ?

The change looks simple, but it would be good for future regressions.  Looks good though!

sbaughman committed:
Refactor conditional for 'hitWordBreak'

sbaughman committed:
Add test for properly separating words after breaks, subscripts, and superscripts
